util/src/main/java/io/kubernetes/client/util/KubeConfig.java

```diff
@@ -214,6 +222,60 @@ public String getAccessToken() {
         }
       }
     }
+    Object exec = currentUser.get("exec");
+    if (exec != null) {
+      // TODO extract to helper method for clarity
+      Map<String, Object> execMap = (Map<String, Object>) exec;
+      // https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins
+      String apiVersion = (String) execMap.get("apiVersion");
+      if ("client.authentication.k8s.io/v1beta1".equals(apiVersion)) { // TODO or v1alpha1 is apparently identical and could be supported
+        String command = (String) execMap.get("command");
+        List<Map<String, String>> env = (List) execMap.get("env");
+        List<String> args = (List) execMap.get("args");
+        // TODO relativize command to basedir of config file (requires KubeConfig to be given a basedir)
+        List<String> argv = new ArrayList<>();
+        argv.add(command);
+        if (args != null) {
+          argv.addAll(args);
+        }
+        ProcessBuilder pb = new ProcessBuilder(argv);
+        if (env != null) {
+          // TODO apply
+        }
+        pb.redirectError(ProcessBuilder.Redirect.INHERIT);
+        try {
+          Process proc = pb.start();
+          JsonElement root = null;
+          try (InputStream is = proc.getInputStream();
+               Reader r = new InputStreamReader(is, StandardCharsets.UTF_8)) {
+            root = new JsonParser().parse(r);
+          } catch (JsonParseException x) {
+            log.error("Failed to parse output of " + command, x);
+          }
+          int r = proc.waitFor();
+          if (r == 0) {
```

Comment:
flip this around for less indents:

```java
if (r != 0) {
   log.error(...)
   return;
}
...
```

Reply:
Yes, when extracting to its own method it is possible to have multiple `return` statements which I find easier to follow.